### PR TITLE
BREAKING CHANGE: gruvbox is now a single module

### PR DESCRIPTION
**Pull request by ellisonleao**
**palette**: Can now be accessed in the main module through `require('gruvbox').palette` 
**get_base_colors**: now a internal function called get_colors. If you want to get the colors, just get the palette directly
**groups**: now moved to main module
**diffs**: fixing diff highlights for multiple plugins

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,16 +18,16 @@ Neovim 0.8.0+
...
 ```lua
 use { "ellisonleao/gruvbox.nvim" }
 ```
 
-Using `lazy.nvim`
+## Using `lazy.nvim`
 
 ```lua
-{ "ellisonleao/gruvbox.nvim", priority = 1000 }
+{ "ellisonleao/gruvbox.nvim", priority = 1000 , setup = true, opts = ...}
```
Comment (polyzen) [inline README.md:30]: There doesn't seem to be a `setup` property, seems you meant to use `config` instead. Also that would be unnecessary as setting `opts` "will imply `Plugin.config()`".

Comment (ellisonleao) [reply README.md:30]: @polyzen actualy it's a typo, using config with opts will basically call `require('gruvbox').setup(opts)`. Will fix it

**Comment by ambroisie** [comment]
It's kind of annoying to have to use `palette` instead of `get_colors`, as it means the end-user has to re-implement the `light`/`dark` logic themselves.

**Comment by darkangel-ua** [comment]
Moreover user has to deal not only with light/dark but also with `contrast` stuff because `contrast` is passed to `setup` which makes all this thing even worse and recursive...

**Pull request merged**